docs/source/notes/broadcasting.rst

```diff
@@ -0,0 +1,108 @@
+.. _broadcasting-semantics:
+
+Broadcasting semantics
+==============
+
+Many PyTorch operations support :any:`NumPy Broadcasting Semantics <numpy.doc.broadcasting>`.
+
+In short, if a PyTorch operation supports broadcast, then its Tensor arguments can be
+automatically expanded to be of equal sizes (without making copies of the data).
```

Comment:
Agreed, and I wasn't consistent in this PR.  In general I prefer "shape" but there is already a size() function and a torch.Size object.  I generally used size/sizes except in cases I thought shape was significantly clearer.